support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -161,32 +184,85 @@ func (r *PodReconciler) updateorCreatePodEntry(ctx context.Context, pod *corev1.
 }
 
 // podSpiffeID returns the desired spiffe ID for the pod, or nil if it should be ignored
-func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) string {
+func (r *PodReconciler) podSpiffeID(pod *corev1.Pod) (string, error) {
 	if r.c.PodLabel != "" {
 		// the controller has been configured with a pod label. if the pod
 		// has that label, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if labelValue, ok := pod.Labels[r.c.PodLabel]; ok {
-			return makeID(r.c.TrustDomain, "%s", labelValue)
+			return makeID(r.c.TrustDomain, "%s", labelValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	if r.c.PodAnnotation != "" {
 		// the controller has been configured with a pod annotation. if the pod
 		// has that annotation, use the value to construct the pod entry. otherwise
 		// ignore the pod altogether.
 		if annotationValue, ok := pod.Annotations[r.c.PodAnnotation]; ok {
-			return makeID(r.c.TrustDomain, "%s", annotationValue)
+			return makeID(r.c.TrustDomain, "%s", annotationValue), nil
 		}
-		return ""
+		return "", nil
 	}
 
 	// the controller has not been configured with a pod label or a pod annotation.
+	if r.c.IdentityTemplate != "" {
+		// create an entry using provided identity template.
+		svid, err := r.getIdentityTemplate(pod)
+		if err != nil {
+			return "", err
+		}
+		return makeID(r.c.TrustDomain, svid), nil
+	}
+
+	// the controller has not been configured with a specific identity format.
 	// create an entry based on the service account.
-	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName)
+	return makeID(r.c.TrustDomain, "ns/%s/sa/%s", pod.Namespace, pod.Spec.ServiceAccountName), nil
 }
 
 func (r *PodReconciler) podParentID(nodeName string) string {
 	return makeID(r.c.TrustDomain, "k8s-workload-registrar/%s/node/%s", r.c.Cluster, nodeName)
 }
+
+func (r *PodReconciler) getIdentityTemplate(pod *corev1.Pod) (string, error) {
+	tpl := r.c.IdentityTemplate
+	tmpl, err := template.New("IdentityTemplate").Parse(tpl)
+	if err != nil {
+		r.c.Log.WithError(err).Errorf("error parsing the template %q", tpl)
+		return "", err
+	}
```

Comment:
Is it possible to do this once, in `NewPodReconciler()` since the template format won't change?

Reply:
Won't change over what timescale?  I'm all for caching things where possible... but presuming they *never* change seems to require some demonstration :)

Reply:
The `k8s-workload-registrar` isn't dynamically configurable, so if you want to change any config you have to restart the container. SPIRE server has the same behavior.

Reply:
@faisal-memon OK.. that seems like a big problem... are you telling me I have to basically reboot my ability to change issue identities whenever I want to reconfigure?  What if a Pod is scheduled while it's rebooting?  How do I operationalize that in a reasonable way?

Reply:
Unfortunately yes. In that case we will get the reconcile event when the k8s-workload-registrar comes back up.

Reply:
@faisal-memon I don't see that as deployable :(  Not by anyone... outage on re-configure isn't going to fly in any real cluster :(

Reply:
There is no outage, everything that was running before will continue to run. New pods deployed while the re-configure is happening will see a delay to get their certs issued. It would be a good enhancement to allow for dynamic reconfig, but out of scope for this PR.

Reply:
@faisal-memon Ah... got it... so degradation, not outage.  Totally agree about it being a different scope to allow dynamic reconfig, not this PR.

Reply:
done